dotcom-rendering/src/components/ShareButton.importable.tsx

```diff
@@ -0,0 +1,229 @@
+import { css } from '@emotion/react';
+import { ArticleDesign } from '@guardian/libs';
+import { palette, until } from '@guardian/source-foundations';
+import type { Size } from '@guardian/source-react-components';
+import {
+	Button,
+	SvgCheckmark,
+	SvgShare,
+} from '@guardian/source-react-components';
+import { useEffect, useState } from 'react';
+import { palette as themePalette } from '../palette';
+import LinkIcon from '../static/icons/link-icon.svg';
+import { transparentColour } from '../lib/transparentColour';
+
+type Props = {
+	size?: Size;
+	pageId: string;
+	blockId?: string;
+	webTitle: string;
+	format: ArticleFormat;
+	context: Context;
+};
+
+type Context = 'ArticleMeta' | 'LiveBlock' | 'SubMeta';
+
+const sharedButtonStyles = (sizeXSmall: boolean, isCopied: boolean) => css`
+	border-color: ${themePalette('--share-button-border')};
+	width: ${sizeXSmall ? '101px' : '132px'};
+	height: ${sizeXSmall ? '24px' : '36px'};
+	padding: ${sizeXSmall && !isCopied && '0 10px'};
+`;
+
+const copiedButtonStyles = (sizeXSmall: boolean) => css`
+	color: ${themePalette('--share-button-copied')};
+	padding: ${sizeXSmall ? '0 4px' : '0 10px'};
+
+	svg {
+		fill: ${palette.success[400]};
+	}
+`;
+
+const buttonStyles = css`
+	color: ${themePalette('--share-button')};
+	svg {
+		fill: ${themePalette('--share-button')};
+	}
+	:hover {
+		background-color: ${themePalette('--share-button')};
+		border-color: ${themePalette('--share-button')};
+		color: ${themePalette('--share-button-hover')};
+		svg {
+			fill: ${themePalette('--share-button-hover')};
+		}
+	}
+`;
+
+const nativeShare = (sizeXSmall: boolean) => css`
+	width: ${sizeXSmall ? '79px' : '105px'};
+	border-color: ${themePalette('--share-button-border')};
+`;
+
+const hoverStyles = css`
+	:hover {
+		color: ${themePalette('--share-button-liveblog-mobile')};
+		background-color: ${palette.neutral[100]};
+		svg {
+			fill: ${themePalette('--share-button-liveblog-mobile')};
```

Comment:
Same as above, if you feel like it:

```suggestion
			fill: currentColor;
```